python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,828 @@
-            for field in schema_definition.structure["schema"]["fields"]:
-                field_name = field["name"]
-                field_type = field["type"]
-                is_list = False
-
-                clean_field_type = field_type
-                if field_type.endswith("[]"):
-                    is_list = True
-                    clean_field_type = field_type[:-2]
-                if "(" in clean_field_type:
-                    clean_field_type = clean_field_type[: clean_field_type.find("(")]
-
-                # let's make sure the type of the field is known, or create it if it's not
-                potential_definitions = find_definitions_by_name(clean_field_type)
-                if len(potential_definitions) != 1:
-                    if len(potential_definitions) == 0:
+        Returns:
+            The field value.
+        """
+        python_type = defining_definition.structure["primitive"]["python_type"]
+        if is_list:
+            if not field_value:
+                if is_required:
+                    raise LanguageError(message=f"Missing required field {field_name}.", location=None)
+                field_value = []
+            else:
+                for item in field_value:
+                    if not isinstance(item, eval(python_type)):
                         raise LanguageError(
-                            f"Could not find AaC definition for type {clean_field_type} while loading {schema_definition.name}",
-                            get_location_str(field_type, schema_definition.lexemes),
+                            message=f"Invalid value for field '{field_name}'.  Expected type '{python_type}', but found '{type(item)}'",
+                            location=self.get_location_str(field_value, lexemes),
                         )
-                    else:
+        else:
+            if not field_value:
+                if is_required:
+                    raise LanguageError(f"Missing required field {field_name}.")
+            else:
+                if "Any" != python_type:
+                    if not isinstance(field_value, eval(python_type)):
                         raise LanguageError(
-                            f"Discovered multiple AaC definitions for type {clean_field_type} while loading {schema_definition.name}.  You may need to add a package name to differentiate.",
-                            get_location_str(field_type, schema_definition.lexemes),
+                            f"Invalid value for field '{field_name}'.  Expected type '{python_type}', but found '{type(field_value)}'",
+                            self.get_location_str(field_value, lexemes),
                         )
+        return field_value
 
-                parsed_definition = potential_definitions[0]
-
-                create_schema_class(parsed_definition)
+    def enum_check(self, is_list: bool, field_value: Any, field_name: str, lexemes: list, defining_definition: Definition) -> list:
```

Comment:
```suggestion
    def enum_field_value_check(self, is_list: bool, field_value: Any, field_name: str, lexemes: list, defining_definition: Definition) -> list:
```
Think these two new check methods should be named more clearly, they are confirming field value content. Make sure to update where they are called for the name change as well.

Reply:
This method should be relocated to be with the `schema_field_value_check` method down at line 654. This method is used within the `create_field_instance` method just after. You should keep related methods near one another.